core/src/main/java/com/linecorp/armeria/server/logging/LoggingServiceBuilder.java

```diff
@@ -0,0 +1,182 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.logging;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Function;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.MoreObjects.ToStringHelper;
+
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.Response;
+import com.linecorp.armeria.common.logging.LogLevel;
+import com.linecorp.armeria.server.Service;
+
+/**
+ * Builds a new {@link LoggingService}.
+ */
+public class LoggingServiceBuilder {
+
+    private static final Function<HttpHeaders, HttpHeaders> DEFAULT_HEADERS_SANITIZER = Function.identity();
+    private static final Function<Object, Object> DEFAULT_CONTENT_SANITIZER = Function.identity();
+
+    private LogLevel requestLogLevel = LogLevel.TRACE;
+    private LogLevel successfulResponseLogLevel = LogLevel.TRACE;
+    private LogLevel failureResponseLogLevel = LogLevel.WARN;
+    private Function<HttpHeaders, HttpHeaders> requestHeadersSanitizer = DEFAULT_HEADERS_SANITIZER;
+    private Function<Object, Object> requestContentSanitizer = DEFAULT_CONTENT_SANITIZER;
+    private Function<HttpHeaders, HttpHeaders> responseHeadersSanitizer = DEFAULT_HEADERS_SANITIZER;
+    private Function<Object, Object> responseContentSanitizer = DEFAULT_CONTENT_SANITIZER;
+    private Sampler sampler = Sampler.ALWAYS_SAMPLE;
+
+    /**
+     * Sets the {@link LogLevel} to use when logging requests. If unset, will use {@link LogLevel#TRACE}.
+     */
+    public LoggingServiceBuilder requestLogLevel(LogLevel requestLogLevel) {
+        this.requestLogLevel = requireNonNull(requestLogLevel, "requestLogLevel");
+        return this;
+    }
+
+    /**
+     * Sets the {@link LogLevel} to use when logging successful responses (e.g., no unhandled exception).
+     * If unset, will use {@link LogLevel#TRACE}.
+     */
+    public LoggingServiceBuilder successfulResponseLogLevel(LogLevel successfulResponseLogLevel) {
+        this.successfulResponseLogLevel =
+                requireNonNull(successfulResponseLogLevel, "successfulResponseLogLevel");
+        return this;
+    }
+
+    /**
+     * Sets the {@link LogLevel} to use when logging failure responses (e.g., failed with an exception).
+     * If unset, will use {@link LogLevel#WARN}.
+     */
+    public LoggingServiceBuilder failureResponseLogLevel(LogLevel failureResponseLogLevel) {
+        this.failureResponseLogLevel = requireNonNull(failureResponseLogLevel, "failureResponseLogLevel");
+        return this;
+    }
+
+    /**
+     * Sets the {@link Function} to use to sanitize request headers before logging. It is common to have the
+     * {@link Function} remove sensitive headers, like {@code Cookie}, before logging.
```

Comment:
Doesn't it need the consistency with the javadoc of requestContentSanitizer? (If unset part is missing~)